src/shared/services/form/field-type-sanitiser.ts

```diff
@@ -13,33 +17,48 @@ export class FieldTypeSanitiser {
    */
    sanitiseLists(caseFields: CaseField[], editForm: any) {
 
-    this.getDynamicListsFromCaseFields(caseFields).forEach(dynamicField => {
-      this.getListOfKeysFromEditForm(editForm).forEach((key) => {
-        this.createValueCodePairAlongWithListIfKeyExistsInForm(dynamicField, key, editForm);
-      });
+    caseFields.forEach(caseField => {
+
+      switch (caseField.field_type.type) {
+        case FieldTypeSanitiser.FIELD_TYPE_DYNAMIC_LIST:
+          this.getListOfKeysFromEditForm(editForm).forEach((key) => {
+            this.createValueCodePairAlongWithListIfKeyExistsInForm(caseField, key, editForm);
+          });
+          break;
+
+        case FieldTypeSanitiser.FIELD_TYPE_COMPLEX:
+          this.sanitiseLists(caseField.field_type.complex_fields, editForm[caseField.id]);
+          break;
+
+        case FieldTypeSanitiser.FIELD_TYPE_COLLECTION:
+          editForm[caseField.id].forEach(formElement => {
+            this.sanitiseLists(caseField.field_type.collection_field_type.complex_fields, formElement.value);
+          });
+          break;
+
+        default:
```

Comment:
Can we simply return a value instead of doing nothing for default case?

Reply:
Function doesn't return anything actually. Do you think it would be OK if I remove the default completely?